encodings/roaring/src/boolean/mod.rs

```diff
@@ -135,9 +133,12 @@ impl IntoCanonical for RoaringBoolArray {
             .to_bitset()
             .ok_or_else(|| vortex_err!("Failed to convert RoaringBitmap to Bitset"))?;
 
-        let buffer = ArrowBuffer::from_slice_ref(bitset.as_slice());
+        let byte_length = (self.len() + 7) / 8;
+        let mut buffer = MutableBuffer::with_capacity(byte_length);
+        buffer.extend_from_slice(bitset.as_slice());
+        buffer.extend_zeros(byte_length - bitset.as_slice().len());
```

Comment:
doesn't `bitset.as_slice()` return a u64?

Reply:
it does return u64, the buffer knows how to take any slice of native arrow type

Reply:
Oh, I see what you mean